### ISSUE TITLE
Release ToolPRMBench artifacts (models, dataset) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [David-Li0406/ToolPRMBench#1]
Hi @David-Li0406 🤗

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2601.12294.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I saw in your GitHub README that you are planning to release the code and data for **ToolPRMBench** soon. It'd be great to make the checkpoints and the benchmark dataset available on the 🤗 hub, to improve their discoverability and visibility within the AI agent community.

## Uploading models

We can add tags so that people find your Process Reward Models when filtering https://huggingface.co/models. See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverage the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

## Uploading dataset

Would be awesome to make the benchmark dataset available on 🤗 , so that people can easily load it:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/ToolPRMBench")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the interaction histories and step-level rewards directly in the browser.

Let me know if you're interested or need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗
